zulipterminal/model.py

```diff
@@ -559,6 +560,50 @@ def update_private_message(self, msg_id: int, content: str) -> bool:
         display_error_if_present(response, self.controller)
         return response["result"] == "success"
 
+    def toggle_topic_resolved_status(self, stream_id: int, topic_name: str) -> bool:
+        if self.initial_data.get("realm_allow_community_topic_editing", None):
+            self.controller.report_error(" Editing topic is disabled")
+        else:
+            request: Dict[str, Any] = {
+                "anchor": "newest",
+                "num_before": 1,
+                "num_after": 0,
+                "narrow": [
+                    {
+                        "operator": "stream",
+                        "operand": self.stream_dict[stream_id]["name"],
+                    },
+                    {"operator": "topic", "operand": topic_name},
+                ],
+            }
+            response = self.client.get_messages(request)
```

Comment:
Can `model.get_messages()` be used here? We may have to force users to narrow to a topic to resolve it so that `self.narrow` is not a problem.

If that is not ideal, would it be more readable if this whole request (along with error handling which you haven't done) was extracted into a function like`get_latest_message_in_topic`?